### PR TITLE
Trivial registry improvements

### PR DESCRIPTION
**Pull request by mkykadir**
See each commit

### CODE REVIEW

**Comment by Geod24** [comment]
CI says no

File: source/agora/node/Registry.d [hunk 1]
```diff
@@ -822,18 +824,28 @@ private struct ZoneData
...
-        if (this.type == ZoneType.primary)
-        {
-            // Serial's value wraps around so the cast is safe
-            const serial = cast(uint) Clock.currTime(UTC()).toUnixTime();
-            this.soa = this.config.fromConfig(this.root, serial);
-        }
-        else
-            // SOA is not default-constructible, so the compiler complains
-            this.soa = SOA.init;
+        this.soa = this.config.fromConfig(this.root);
```
Comment (Geod24) [inline source/agora/node/Registry.d]: This is what is causing the test failure. If the zone is not primary, we shouldn't read the SOA from the config file.

Comment (mkykadir) [reply source/agora/node/Registry.d]: That would be the next cause, but currently main cause is `ZoneType` only includes `primary` and that's the default one.

Comment (mkykadir) [reply source/agora/node/Registry.d]: So, I needed to disable `post...` endpoint if Zone is not primary and test needed to be updated to initialize a primary Validator zone.

Comment (Geod24) [reply source/agora/node/Registry.d]: I'm not sure why ? We can fwd `post`, no ?

Comment (mkykadir) [reply source/agora/node/Registry.d]: Secondary is not fully implemented in this PR, this only makes it easier to detect primary and non-primary zones through type field. Thus there is no fwd configuration available. Main reason behind is me allowing `onAcceptedBlock` to only `primary` zone, if I revert that, then there is no need to disable `post...`s

Comment (Geod24) [reply source/agora/node/Registry.d]: Yeah that makes sense.

**Comment by Geod24** [comment]
```
2022-01-25 07:53:01,584 Info [agora.network.Manager] - Couldn't register our address: geod24.LocalRest.ClientException@submodules/localrest/source/geod24/LocalRest.d-mixin-1217(1236): Request to void geod24.LocalRest.RemoteAPI!(NameRegistryAPI, Serializer).RemoteAPI.postValidator(RegistryPayload _param_0) @safe couldn't be processed : agora.common.Ensure.FormattedException@source/agora/node/Registry.d(229): Couldn't register, server is not primary for the zone
----------------
source/agora/common/Ensure.d:111 @safe void agora.common.Ensure.ensure!().ensure(bool, immutable(char)[], immutable(char)[], int) [0x1027e68d2]
source/agora/node/Registry.d:229 @safe void agora.node.Registry.NameRegistry.postValidator(agora.api.Registry.RegistryPayload) [0x102ea6746]
source/agora/test/Base.d:2493 @safe void agora.test.Base.RegistryNode.postValidator(agora.api.Registry.RegistryPayload) [0x102ea66d7]
submodules/localrest/source/geod24/LocalRest.d-mixin-724:741 void geod24.LocalRest.RemoteAPI!(agora.test.Base.FullRegistryAPI, agora.test.Base.Serializer).RemoteAPI.handleCommand(geod24.LocalRest.Command, agora.test.Base.FullRegistryAPI, geod24.LocalRest.FilterAPI, geod24.concurrency.Channel!(geod24.LocalRest.Response).Channel) [0x102edf2d4]
submodules/localrest/source/geod24/LocalRest.d:864 void geod24.LocalRest.RemoteAPI!(agora.test.Base.FullRegistryAPI, agora.test.Base.Serializer).RemoteAPI.spawned!(agora.test.Base.RegistryNode).spawned(geod24.concurrency.Channel!(geod24.LocalRest.Connection).Channel, immutable(char)[], int, agora.node.Config.Config, shared(geod24.Registry.AnyRegistry)*, immutable(agora.consensus.data.Block.Block)[], in ref agora.test.Base.TestConf, shared(ulong)*, geod24.concurrency.Channel!(agora.test.Base.DNSQuery).Channel).runNode().__lambda4().__lambda14() [0x102eee651]
core/thread/context.d:46 void core.thread.context.Callable.opCall() [0x103ccf2dc]
core/thread/fiber.d:1151 void core.thread.fiber.Fiber.run() [0x103ccf468]
core/thread/fiber.d:198 fiber_entryPoint [0x103ccf3a5]
```

File: source/agora/test/Base.d [hunk 2]
```diff
@@ -2312,7 +2313,19 @@ public APIManager makeTestNetwork (APIManager : TestAPIManager = TestAPIManager)
       network : net.nodes.map!(pair => "http://"~pair.address).array.idup,
       logging: test_conf.logging,
       event_handlers : test_conf.event_handlers,
-      registry: { enabled: true, },
+      registry: {
+          enabled: true,
+          validators: {
+              authoritative: true,
+              primary: "name.registry",
+              soa: { email: SetInfo!string("test@testnet"), },
```
Comment (Geod24) [inline source/agora/test/Base.d]:
```suggestion
              soa: { email: SetInfo!string("test@testnet", true), },
```

Same below

**Review by Geod24 — APPROVED** [review]

**Pull request merged**